Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts

```diff
@@ -19,4 +36,108 @@ export class CalendarComponent {
       right: 'dayGridMonth,timeGridWeek,timeGridDay'
     }
   };
+
+
+  constructor(public dialog: MatDialog,
+              private toastr: ToastrService,
+              private calendarService: CalendarService) {
+  }
+
+  ngOnInit() {
+    this.calendarService.getEvents()
+      .subscribe(events  => {
+        this.events = events;
+        this.calendarOptions.events = this.events;
+      });
+  }
+
+  ngAfterViewInit() {
+    this.calendarComponentApi = this.calendarComponent.getApi();
+  }
+
+  onOpenDialog(): void {
+    const dialogRef = this.dialog.open(EventModalComponent, {
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        this.calendarComponentApi.addEvent(result);
+      }
+    });
+  }
+
+  dateClick(calendarEvent) {
+    const dialogRef = this.dialog.open(EventModalComponent, {
+      data: {
+        date: calendarEvent.dateStr
+      }
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        this.calendarComponentApi.addEvent(result);
+      }
+    });
+  }
+
+  editEventFromDialog(calendarEvent) {
+    const isDeleteEvent = calendarEvent.deleteEvent;
+    const event = this.calendarComponentApi.getEventById(calendarEvent.id);
+
+    if (event && !isDeleteEvent) {
+      event.setProp('title', calendarEvent.title);
+      event.setAllDay(calendarEvent.allDay);
+      event.setExtendedProp('description', calendarEvent.description);
+      event.setExtendedProp('location', calendarEvent.location);
+      event.setStart(calendarEvent.start);
+      event.setEnd(calendarEvent.end);
+    }
+
+  }
+
+  editEventFromCalendar(event: EventApi) {
+    const updatedEventStartStr = event.event.startStr;
+    const oldEventStartStr = event.oldEvent.startStr;
+    const updatedEventEndStr = event.event.endStr;
+    const oldEventEndStr = event.oldEvent.endStr;
+    if (updatedEventStartStr !== oldEventStartStr || updatedEventEndStr !== oldEventEndStr) {
+
+      const eventChange: CalendarEventForm = {
+        id: event.event.id,
+        title: event.event.title,
+        start: event.event.startStr,
+        end: event.event.endStr === '' ? null : event.event.endStr,
+        location: event.event.extendedProps.location,
+        description: event.event.extendedProps.description,
+        allDay: event.event.allDay
+      };
+
+      this.calendarService.editEvent(eventChange)
+        .subscribe(result => {
+        });
+    }
+
+  }
+
+  editEvent(info) {
+    const dialogEditRef = this.dialog.open(EditEventComponent, {
+      data: info.event
+    });
+
+    dialogEditRef.afterClosed().subscribe(result => {
```

Comment:
Use `OnDestroy` event to unsubscribe from Observable

Reply:
fixed